### PR TITLE
[testsuite] Add retries to test_startup_sync_state

### PR DESCRIPTION
**Pull request by ankushagarwal**
# Summary

test_startup_sync_state was flaky probably because we were
not waiting for sync to complete.

This adds logic to retry and verify that the balances that
we expect are the actual balances in the wallet

#Test Plan

Test succeeds locally

### CODE REVIEW

**Comment by andll** [comment]
That probably should not be prefixed with [cluster_test]

File: testsuite/tests/libratest/smoke_test.rs [hunk 1]
```diff
@@ -360,6 +361,20 @@ fn test_startup_sync_state() {
     // during a node startup
     fs::remove_dir_all(state_db_path).unwrap();
     assert!(env.validator_swarm.add_node(peer_to_stop, false).is_ok());
+    retry(Fixed::from_millis(10_000).take(5), || {
```
Comment (andll) [inline testsuite/tests/libratest/smoke_test.rs]: Lets rescale it to retry more often so test pass faster? something like 50 times every second wait?

Comment (ankushagarwal) [reply testsuite/tests/libratest/smoke_test.rs]: 👍

**Comment by ankushagarwal** [comment]
We already have added this as a dependency for `cluster-test`[1] . We not only use the `retry` function, but also the `Fixed` delay iterator [2].

I am not sure where to draw the line for adding a dependency v/s rolling our our own module. In this case I think it's reasonable to add this dependency as retries are very common logic which I expect to be used in various parts of the codebase and the `retry` dependency adds value here.

[1] https://github.com/libra/libra/blob/8e2375914339e61dd6a51c77b49aaf30e93edd39/testsuite/cluster-test/Cargo.toml#L18

[2] https://github.com/jimmycuadra/retry/blob/ab48d2b467b04300414bf8cb0063ac074df43632/src/delay.rs#L107

**Comment by andll** [comment]
I generally of agree with @huitseeker tbh.
I did not push for it in cluster test, but if we could have retry as our own function instead of pulling dependency, I would like it more. However to me this is not a blocker, so I'll leave it to you guys

**Comment by mimoo** [comment]
Agree, we need to stop pulling non-std dependencies that are minor.

BTW I'm surprised we didn't get a warning from some bot, @metajack  ?

**Comment by metajack** [comment]
Yes, cargo-dep-bot should have commented on this. I'll see if I can find the logs tomorrow.

**Comment by mimoo** [comment]
btw thx for addressing this @ankushagarwal

**Pull request closed without merging**